app/javascript/components/loadingIcon.jsx

```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import {faDiceFive} from "@fortawesome/free-solid-svg-icons";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+
+const LoadingIcon = () => {
+    const renderLoadingIcon = () => {
+    return(
+    
+        <FontAwesomeIcon icon={faDiceFive} size="3x" spin/>
+    
+    );  
+    }   
+
+
+    return(
+        <div className="loading-icon">
+            {renderLoadingIcon()}
```

Comment:
Wouldn't it be more clear to put the component in `return` rather than making a function out of it? If unnecessary just do:
```js
<div className="loading-icon">
   <FontAwesomeIcon icon={faDiceFive} size="3x" spin/>
</div>
```